ql/src/java/org/apache/hadoop/hive/ql/exec/spark/SparkReduceRecordHandler.java

```diff
@@ -132,10 +133,12 @@ public void init(JobConf job, OutputCollector output, Reporter reporter) throws
     isTagged = gWork.getNeedsTagging();
     try {
       keyTableDesc = gWork.getKeyDesc();
-      inputKeyDeserializer = ReflectionUtils.newInstance(keyTableDesc
-        .getDeserializerClass(), null);
-      SerDeUtils.initializeSerDe(inputKeyDeserializer, null, keyTableDesc.getProperties(), null);
-      keyObjectInspector = inputKeyDeserializer.getObjectInspector();
+      AbstractSerDe serde = ReflectionUtils.newInstance(keyTableDesc
+        .getSerDeClass(), null);
+      serde.initialize(null, keyTableDesc.getProperties(), null);
+      keyObjectInspector = serde.getObjectInspector();
+
+      inputKeyDeserializer = serde;
```

Comment:
Please call this inputKeySerDe. Also no need to create a variable called serde for an interim time.

Reply:
Done.